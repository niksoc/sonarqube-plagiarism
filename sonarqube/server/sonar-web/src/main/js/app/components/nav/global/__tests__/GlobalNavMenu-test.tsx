/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import GlobalNavMenu from '../GlobalNavMenu';

it('should work with extensions', () => {
  const appState = {
    globalPages: [{ key: 'foo', name: 'Foo' }],
    qualifiers: ['TRK']
  };
  const currentUser = {
    isLoggedIn: false
  };
  const wrapper = shallow(
    <GlobalNavMenu appState={appState} currentUser={currentUser} location={{ pathname: '' }} />
  );
  expect(wrapper.find('Dropdown')).toMatchSnapshot();
});

it('should show administration menu if the user has the rights', () => {
  const appState = {
    canAdmin: true,
    globalPages: [],
    qualifiers: ['TRK']
  };
  const currentUser = {
    isLoggedIn: false
  };
  const wrapper = shallow(
    <GlobalNavMenu appState={appState} currentUser={currentUser} location={{ pathname: '' }} />
  );
  expect(wrapper).toMatchSnapshot();
});
