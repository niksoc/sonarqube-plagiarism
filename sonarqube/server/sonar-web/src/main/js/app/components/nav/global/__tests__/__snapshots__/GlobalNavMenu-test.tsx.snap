// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show administration menu if the user has the rights 1`] = `
<ul
  className="global-navbar-menu"
>
  <li>
    <Link
      className=""
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/projects"
    >
      projects.page
    </Link>
  </li>
  <li>
    <Link
      className=""
      onlyActiveOnIndex={false}
      style={Object {}}
      to={
        Object {
          "pathname": "/issues",
          "query": Object {
            "resolved": "false",
          },
        }
      }
    >
      issues.page
    </Link>
  </li>
  <li>
    <Link
      activeClassName="active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/coding_rules"
    >
      coding_rules.page
    </Link>
  </li>
  <li>
    <Link
      activeClassName="active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/profiles"
    >
      quality_profiles.page
    </Link>
  </li>
  <li>
    <Link
      activeClassName="active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to={
        Object {
          "pathname": "/quality_gates",
        }
      }
    >
      quality_gates.page
    </Link>
  </li>
  <li>
    <Link
      activeClassName="active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/admin"
    >
      layout.settings
    </Link>
  </li>
</ul>
`;

exports[`should work with extensions 1`] = `
<Dropdown
  overlay={
    <ul
      className="menu"
    >
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/extension/foo"
        >
          Foo
        </Link>
      </li>
    </ul>
  }
  tagName="li"
>
  <Component />
</Dropdown>
`;
